### PR TITLE
Improvements based on review feedback

### PR DESCRIPTION
**Pull request by yorinasub17**
- Extract out `$1` into a named variable
- Add back in `set -e`
- Call `contains_element` as part of the if.
- Add comment about why this is doing what it is doing
- Use `! -z` to check for debug setting
- Debug logs to stderr
- Change logic around `check_changed_file`

### CODE REVIEW

File: hooks/helmlint.sh [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
 
-# We don't set `set -e` in this script because some of the functions will return a non-zero value (e.g contains_element)
+set -e
+
+# Run helm lint on the chart path.
+# helm lint can only be run on the root path of a chart, so this pre-commit hook will take the changed files and resolve
+# it to the helm chart path. The helm chart path is determined by a heuristic: it is the directory containing the
+# Chart.yaml file.
+# Note that pre-commit will only feed this the files that changed in the commit, so we can't do the filtering at the
+# hook setting level (e.g `files: Chart.yaml` will not work if no changes are made in the Chart.yaml file).
```
Comment (brikis98) [inline hooks/helmlint.sh]: But do you need to run the linter if some other file besides `Chart.yaml` changed?

File: hooks/helmlint.sh [hunk 2]
```diff
@@ -29,75 +33,70 @@ contains_element() {
...
+# An array to keep track of which charts we already linted
+seen_chart_paths=()
 
 for file in "$@"; do
   debug "Checking $file"
-  check_changed_file "$file"
+  file_chart_path=$(chart_path "$file")
+  debug "Resolved $file to chart path $file_chart_path"
+
+  if [[ ! -z $file_chart_path ]]; then
```
Comment (brikis98) [inline hooks/helmlint.sh]: Wrap variable in quotes

File: hooks/helmlint.sh (hunk 1 above)
Comment (yorinasub17) [reply hooks/helmlint.sh]: Yes. To explain, here is what a typical helm chart directory structure looks like:

```
└── root
    ├── README.md
    ├── Chart.yaml
    ├── charts
    │   └── postgres-9.5.6.tar.gz
    └── templates
        ├── deployment.yaml
        ├── service.yaml
        └── _helpers.tpl
```

The `Chart.yaml` is metadata that helm uses to index the chart, and is added to the release info. It includes things like `name`, `version`, and `maintainers`.

The `charts` directory are subcharts / dependencies that are deployed with the chart.

The `templates` directory is what contains the go templates to render the Kubernetes resource yaml. Also includes the template definitions (suffix `.tpl`).

The linter runs on the `root` and checks the templates to make sure it is well formed. This includes syntax checks as well as checking for undefined template functions amongst other things.

Given that, I had to do this search for the `Chart.yaml` to find the root of the chart from one of the template files changing, as that is more likely to have changed in the commit than the `Chart.yaml` metadata file.

I hope that makes sense!

Comment (brikis98) [reply hooks/helmlint.sh]: Ohhhh. OK, I'm with you now. Thanks for the context. Consider adding this as a comment to the code or to the README!

**Comment by yorinasub17** [comment]
Ok added the comment regarding why we need the directory lookup. Merging and releasing!

**Pull request merged**